src/lib/integration_test_local_engine/docker_compose.ml

```diff
@@ -0,0 +1,63 @@
+open Core
+
+module Compose = struct
+  module DockerMap = struct
+    type 'a t = (string, 'a, String.comparator_witness) Map.t
+
+    let empty = Map.empty (module String)
+  end
+
+  module Service = struct
+    module Volume = struct
+      type t = {type_: string; source: string; target: string}
+
+      let create name =
+        {type_= "bind"; source= "." ^/ name; target= "/root" ^/ name}
+
+      let to_yojson {type_; source; target} =
```

Comment:
You shouldn't need a custom deriver for this. `[@@deriving to_yojson]` should work on this type def if you use the `[@key "type"]` decorator for the `type_` field (see https://github.com/ocaml-ppx/ppx_deriving_yojson#key).